src/engine/naive_engine.cc

```diff
@@ -165,14 +178,22 @@ class NaiveEngine final : public Engine {
     }
     CHECK(this->req_completed_)
         << "NaiveEngine only support synchronize Push so far";
+    // increment var version
+    for (auto var : mutable_vars) {
```

Comment:
In case of deletevariable, will this increment the var version after it is deleted ?

Reply:
Good catch. Should move this before `exec_fun` runs.